pkg/collector/corechecks/containers/containerd.go

```diff
@@ -336,12 +345,28 @@ func parseAndSubmitMem(metricName string, sender aggregator.Sender, stat *v1.Mem
 
 }
 
-func computeCPU(sender aggregator.Sender, cpu *v1.CPUStat, tags []string) {
+func computeCPU(sender aggregator.Sender, cpu *v1.CPUStat, cpuLimits *specs.LinuxCPU, startTime, currentTime time.Time, tags []string) {
+	if cpu == nil || cpu.Usage == nil {
+		return
+	}
+
 	sender.Rate("containerd.cpu.system", float64(cpu.Usage.Kernel), "", tags)
 	sender.Rate("containerd.cpu.total", float64(cpu.Usage.Total), "", tags)
 	sender.Rate("containerd.cpu.user", float64(cpu.Usage.User), "", tags)
-	sender.Rate("containerd.cpu.throttled.periods", float64(cpu.Throttling.ThrottledPeriods), "", tags)
 
+	if cpu.Throttling != nil {
+		sender.Rate("containerd.cpu.throttled.periods", float64(cpu.Throttling.ThrottledPeriods), "", tags)
+		sender.Rate("containerd.cpu.throttled.time", float64(cpu.Throttling.ThrottledTime), "", tags)
+	}
```

Comment:
should we return the values "0" in throttling is `nil`?

Reply:
Previous behavior was to send nothing (incl. `containerd.cpu.usage` if `Throttling` is nil). It did not make sense to me, so I allowed to send CPU Usage anyway and kept previous behavior for `Throttling`.

About `Throttling` itself, values inside the struct are gauges (ever increasing), so if we have value once, we should always have values. So I believe, current behavior is correct.